### PR TITLE
[POC] Add signature verification for plugins

### PR DESCRIPTION
**Pull request by vurusovs**
### Details:
- Add signature verification (for Win) before plugin loading
- Self-signed library can be loaded successfully with current implementation. To reproduce lib signing, use https://stackoverflow.com/questions/84847/how-do-i-create-a-self-signed-certificate-for-code-signing-on-windows

### Tickets:
 - 96205

### CODE REVIEW

File: src/common/util/src/os/win/win_shared_object_loader.cpp [hunk 1]
```diff
@@ -70,9 +70,74 @@
...
+
+// #include <wincrypt.h>
+// #include <wintrust.h>
+
+// #pragma comment (lib, "wintrust")
+
+namespace {
+bool verify_embedded_signature(LPCWSTR pwszSourceFile)
+{
+	GUID WintrustVerifyGuid = WINTRUST_ACTION_GENERIC_VERIFY_V2;
```
Comment (vurusovs) [inline src/common/util/src/os/win/win_shared_object_loader.cpp:84]: Add links:
- https://github.com/konstantin89/windows-pe-signature-verifying/blob/master/pe-signature-utils/src/native_api_wrappers/TrustVerifyWrapper.cpp#L21
- https://learn.microsoft.com/en-us/windows/win32/seccrypto/example-c-program--verifying-the-signature-of-a-pe-file

File: src/common/util/include/openvino/util/shared_object.hpp [hunk 2]
```diff
@@ -21,6 +21,7 @@ namespace util {
  * @param path Full or relative path to the plugin library
  * @return Reference to shared object
  */
+std::shared_ptr<void> load_shared_object(const char* path, const bool& verify_signature);
```
Comment (rkazants) [inline src/common/util/include/openvino/util/shared_object.hpp:24]: I think we need to update description with new param:)

Comment (vurusovs) [reply src/common/util/include/openvino/util/shared_object.hpp:24]: @rkazants Thanks! For the moment it's just a POC to verify CI lib signs are compatible with Runtime signs verification

File: src/common/util/src/os/win/win_shared_object_loader.cpp [hunk 3]
```diff
@@ -70,9 +70,74 @@
...
+bool verify_embedded_signature(const char* path) {
+    return verify_embedded_signature(ov::util::string_to_wstring(path).c_str());
+}
+}  // namespace
 
 namespace ov {
 namespace util {
+std::shared_ptr<void> load_shared_object(const char* path, const bool& verify_signature) {
+    if (verify_signature) {
+        if (!is_absolute_file_path(path)) {
```
Comment (vurusovs) [inline src/common/util/src/os/win/win_shared_object_loader.cpp:126]: Add new field `verify_signature` to `PluginDescriptor` in order to check signs only for required plugins

**Comment by vurusovs** [comment]
Self-signed library can be loaded successfully with current implementation. To reproduce lib signing, use https://stackoverflow.com/questions/84847/how-do-i-create-a-self-signed-certificate-for-code-signing-on-windows

File: src/common/util/src/os/win/win_shared_object_loader.cpp [hunk 4]
```diff
@@ -124,6 +189,23 @@ std::shared_ptr<void> load_shared_object(const char* path) {
 }
 
 #ifdef OPENVINO_ENABLE_UNICODE_PATH_SUPPORT
+std::shared_ptr<void> load_shared_object(const wchar_t* path, const bool& verify_signature) {
+    if (verify_signature) {
+        if (!is_absolute_file_path(ov::util::wstring_to_string(path))) {
+            // TODO: check how it works with file names
+            std::stringstream ss;
+            ss << "Cannot verify signature of library '" << path << "': path isn't absolute.";
```
Comment (vurusovs) [inline src/common/util/src/os/win/win_shared_object_loader.cpp:197]: add `wstring_to_string` for correct message

**Comment by vurusovs** [comment]
For the moment we decided to decline signature verification because initial problem which we are trying to solve is covered by OS through the differentiation of user access rights (more details in a ticket)

**Pull request closed without merging**